src/v/serde/protobuf/tests/parser_test.cc

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+// TODO: Fix bazelbuild/bazel#4446
+#include "bytes/iobuf.h"
+#include "bytes/iobuf_parser.h"
+#include "gtest/gtest.h"
+#include "serde/protobuf/parser.h"
+#include "src/v/serde/protobuf/tests/test_messages_edition2023.pb.h"
```

Comment:
i also encountered having to have a path like this when building pandaproxy with bazel. anyway around it, or just not worth the trouble?

Reply:
The best thing is to fix bazelbuild/bazel#4446

Alternatives:
1. Write our own rule for generating protos for C++ and then we have full control over the library target
2. Do some `-quote` hackery within `copts` for these libraries
3. Reroot all protos to the root of the repo.

If it's worth it is a question of what we want to do long term with the repo. We had once flirted with the idea of moving all of `/src/v` to just `/v` in this repo, and adding the `v/` to include paths everywhere. If we want to do that then maybe the answer is to do nothing for now. If that's not going to be until we stop supporting CMake entirely (which would be when v25.1.x is EOL?) then maybe we should figure out something a little better in the short term.

FWIW I think option 1 isn't huge lift - I could probably even find prior art.

Reply:
>FWIW I think option 1 isn't huge lift - I could probably even find prior art.

the genrule itself would be trivial, getting access to all of the google protos that ship in the tree would be more work. i feel like i could make this work with the cmake_foreign_cc pretty easily at this point. if you happen to have some prior art reference that's cool too.

> If we want to do that then maybe the answer is to do nothing for now.

i could see people being on board with something like re-rooting everything to `/core/..` or `/v` or whatever. `src/` prefix is a bit redundant. what do bigger mono repos do?

Reply:
> what do bigger mono repos do?

I think in "real" monorepo for all our code - we'd see structure like this (ignoring the legacy `v` prefix for vectorized that is nice and short but outdated):

Basically it would be a strange mash of organizational structure + products + code structure. Maybe down the road `core/redpanda` should just have top level folders for each team `{storage,replication,enterprise,performance}` so it's clear who owns what. I know the actual lines are more blurry and we don't actually want to draw that hard of a line, but it's an option.

```
.
├── bazel/
│   └── ...
├── cloud/
│   ├── v2/
│   │   └── ...
│   └── v1/
│       └── ...
├── k8s/
│   ├── helm/
│   │   └── ...
│   └── operator/
│       ├── v1/
│       │   └── ...
│       └── v2/
│           └── ...
├── rpk/
│   ├── go.mod
│   ├── cli/
│   │   └── ...
│   └── pkg/
│       └── ...
└── core/
    ├── transform_sdks/
    │   ├── go/
    │   │   └── ...
    │   ├── rust/
    │   │   └── ...
    │   ├── cpp/
    │   │   └── ...
    │   └── js/
    │       └── ...
    ├── test_util/
    │   └── message_generator/
    │       └── ...
    ├── integration/ # Maybe this should live inside `redpanda` dir
    │   ├── ducktape/
    │   │   └── ...
    │   └── etc...
    └── redpanda/
        ├── main.cc
        ├── bytes/
        │   └── ...
        ├── admin/
        │   └── ...
        ├── kafka/
        │   └── ...
        ├── cloud_storage/
        │   └── ...
        ├── s3_client/
        │   └── ...
        └── etc...
```

Reply:
> getting access to all of the google protos that ship in the tree would be more work.

Via Bazel they expose all the google well known protos as files so we could plug into our actual rule.

https://github.com/protocolbuffers/protobuf/blob/cb7111914918b2c0783f2d44a18a08b6c4b51c4e/BUILD.bazel#L165-L169

also we probably want to use a precompiled binary for `protoc` (there is a flag for this) because it's silly to recompile `protoc` between `-c {dbg,fastbuild,opt}` - Google itself checks in protoc to version control and does something similar.

Reply:
Captured the pre-compiled protoc improvement here https://redpandadata.atlassian.net/browse/CORE-6794